src/pipeline_migration/actions/migrate.py

```diff
@@ -842,6 +846,62 @@ def _resolve_task(self, bundle_upgrade: TaskBundleUpgrade) -> None:
             bundle_upgrade.migrations.append(tb_migration)
 
 
+def has_migration_image(image_repo: str) -> bool:
+    """Guess if an image repository has migration images
+
+    During the transition to decentralized task repositories, not all tasks are built by the tekton
+    bundle builder pipeline and released via release pipeline. This method guesses whether a
+    repository has at least one migration image tag.
+
+    Tags are queried by a pattern that has fixed prefix ``migration-``. Although Quay does SQL LIKE
+    to match tag names, it should be good enough to get existing migration image tags if there is.
+
+    :param image_repo: bundle repository.
+    :type image_repo: str
+    :return: True if migration image tags are retrieved from the given repository. Otherwise, False
+        is returned.
+    """
+    c = Container(image_repo)
+    limit = 10
+    tags_iter = list_active_repo_tags(
+        c, tag_name_pattern=MIGRATION_IMAGE_TAG_LIKE_PATTERN, limit=limit
+    )
+    results: list[bool] = []
+    for i in range(limit):
+        try:
+            tag_name = next(tags_iter)["name"]
+            results.append(MigrationImageTag.parse(tag_name) is not None)
+        except StopIteration:
+            break
```

Comment:
nitpick: instead of a `for i in range(...)` loop and catching `StopIteration`, wouldn't it be simpler to do `for tag in tags_iter`?

Reply:
This `limit` is the quay `listRepoTags` API `limit` argument, it's the number of tags per page.